src/main/java/com/okta/tools/OktaAwsCliAssumeRole.java

```diff
@@ -15,937 +15,67 @@
  */
 package com.okta.tools;
 
-import com.amazonaws.auth.AWSCredentialsProvider;
-import com.amazonaws.auth.AWSStaticCredentialsProvider;
-import com.amazonaws.auth.BasicAWSCredentials;
-import com.amazonaws.auth.BasicSessionCredentials;
-import com.amazonaws.regions.Regions;
-import com.amazonaws.services.securitytoken.AWSSecurityTokenService;
-import com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClientBuilder;
 import com.amazonaws.services.securitytoken.model.AssumeRoleWithSAMLRequest;
 import com.amazonaws.services.securitytoken.model.AssumeRoleWithSAMLResult;
-import com.okta.tools.aws.settings.Configuration;
-import com.okta.tools.aws.settings.Credentials;
-import com.okta.tools.aws.settings.MultipleProfile;
-import com.okta.tools.aws.settings.Profile;
-import com.okta.tools.saml.*;
-import org.apache.commons.lang.StringUtils;
-import org.apache.http.HttpStatus;
-import org.apache.http.StatusLine;
-import org.apache.http.client.CookieStore;
-import org.apache.http.client.entity.UrlEncodedFormEntity;
-import org.apache.http.client.methods.CloseableHttpResponse;
-import org.apache.http.client.methods.HttpGet;
-import org.apache.http.client.methods.HttpPost;
-import org.apache.http.cookie.Cookie;
-import org.apache.http.entity.StringEntity;
-import org.apache.http.impl.client.BasicCookieStore;
-import org.apache.http.impl.client.CloseableHttpClient;
-import org.apache.http.impl.client.HttpClients;
-import org.apache.http.impl.cookie.BasicClientCookie;
-import org.apache.http.message.BasicNameValuePair;
+import com.okta.tools.helpers.ConfigHelper;
+import com.okta.tools.helpers.ProfileHelper;
+import com.okta.tools.helpers.RoleHelper;
+import com.okta.tools.helpers.SessionHelper;
+import com.okta.tools.models.Profile;
+import com.okta.tools.models.Session;
+import com.okta.tools.saml.OktaSaml;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
-import org.json.JSONArray;
-import org.json.JSONException;
-import org.json.JSONObject;
-import org.jsoup.Jsoup;
-import org.jsoup.nodes.Document;
-import org.jsoup.select.Elements;
 
-import java.io.*;
-import java.nio.charset.StandardCharsets;
-import java.nio.file.Files;
-import java.nio.file.Paths;
-import java.nio.file.Path;
+import java.io.IOException;
 import java.time.Instant;
 import java.time.temporal.ChronoUnit;
-import java.util.*;
-import java.util.stream.Collectors;
+import java.util.Optional;
 
 final class OktaAwsCliAssumeRole {
 
     private static final Logger logger = LogManager.getLogger(OktaAwsCliAssumeRole.class);
-    private static final String OKTA_AWS_CLI_SESSION_FILE = ".okta-aws-cli-session";
```

Comment:
👍 While this is technically a breaking change, it's in the context of a single generated session and I prefer your change of moving this into the **.okta** folder.